server/persistence/login.go

```diff
@@ -14,6 +15,22 @@ import (
 	"github.com/offen/offen/server/keys"
 )
 
+func (p *persistenceLayer) LoginSSO(email, salt string) (LoginResult, error) {
+	sha512Hash := sha512.New()
+	sha512Hash.Write([]byte(email))
+	sha512Hash.Write([]byte(salt))
```

Comment:
`Write` could potentially return an error, so this should be checked and handled as well.

Reply:
It can not. see `crypto/sha512/sha512.go:256`

Reply:
Even if the current implementation does never assign a value to `err`, the signature still tells us it could do so at any time.

I think it makes sense to treat the (stable) signature as the contract, not the (unstable) current implementation.

I do know this is a merely theoretical concern, but I think it makes sense to check errors if the signature tells us an error could be returned. Else, you'd need to start having intricate knowledge about all your dependencies and how they actually implement the things they are supposed to do.

Reply:
While I agree in principle, in this case the only reason it would fail was if there were issues in writing to memory. And if that happens, you have bigger issues. The error return is simply part of the `io.Writer` interface, and thus required to satisfy it. Hashing static data isn't something that can "fail".